### DISCUSSION TITLE
Extracting a ZIP inside the ZIP file.

### DISCUSSION THREAD

**Discussion by Xelbayria** [ScoopInstaller/Main#5627] [Q&A] [answered]
Hello, 

I've tried to find a way to use `dir_extract` but I am not sure if this is the right approach to take. 

The structure of the ZIP file is: `download.zip -> $PLUGINSDIR\app-64.7z` where **->** meant "you are inside the content of ZIP file". 

what's the correct approach to extracting the contents from download.zip?

**Comment by amreus** [comment, marked as answer]
@Xelbayria - From another thread I believe you found the answer. 

For anyone else looking you can view the `upscayl` manifest for a good example.

Also, here is a link to a ScoopInstaller search for `app-64.7z` which shows a number of manifests that are extracting the app-64.7z from in a NSIS installer:

https://github.com/search?q=org%3AScoopInstaller%20app-64.7z&type=code

**Comment by Xelbayria** [comment]
ah. I didn't think of using a search like that.  `github.com/search`. This is helpful for me to find other examples. Thanks for replying.
